plugin/core/windows.py

```diff
@@ -510,35 +508,6 @@ def discard(self, window: sublime.Window) -> None:
         self._windows.pop(window.id(), None)
 
 
-opening_files = {}  # type: Dict[str, Tuple[Promise, Callable[[Optional[sublime.View]], None]]]
-
-
-def open_file(window: sublime.Window, file_path: str, flags: int = 0, group: int = -1) -> Promise:
-    """Open a file asynchronously. It is only safe to call this function from the UI thread."""
-    view = window.open_file(file_path, flags, group)
-    if not view.is_loading():
-        # It's already loaded. Possibly already open in a tab.
-        return Promise.resolve(view)
-
-    # Is the view opening right now? Then return the associated unresolved promise
-    for fn, value in opening_files.items():
-        if fn == file_path or os.path.samefile(fn, file_path):
-            # Return the unresolved promise. A future on_load event will resolve the promise.
-            return value[0]
-
-    # Prepare a new promise to be resolved by a future on_load event (see the event listener in main.py)
-    def fullfill(resolve: ResolveFunc) -> None:
-        global opening_files
-        # Save the promise in the first element of the tuple -- except we cannot yet do that here
-        opening_files[file_path] = (None, resolve)  # type: ignore
-
-    promise = Promise(fullfill)
-    tup = opening_files[file_path]
-    # Save the promise in the first element of the tuple so that the for-loop above can return it
-    opening_files[file_path] = (promise, tup[1])
-    return promise
```

Comment:
This had to go back to promise.py, because I got entangled in circular imports.

Reply:
.edit imports .windows.open_file
.windows import .sessions.get_plugin
.sessions imports .edit.apply_workspace_edit
...